### PR TITLE
Support SSL

### PR DESCRIPTION
**Pull request by c910335**
### Description of the Change

Implemented #151.

#### Usage

config/application.cr

```crystal
Amber::Server.instance.config do |app|
  # ...
  app.ssl_key_file = "config/privkey.pem"
  app.ssl_cert_file = "config/fullchain.pem"
end
```

### Why Should This Be In Core?

SSL is important for every website, but it is too much work with Nginx or Apache or something like them.

### Benefits

Start with SSL easily.

### CODE REVIEW

File: src/amber.cr [hunk 1]
```diff
@@ -89,7 +91,9 @@ module Amber
 
     def start
       time = Time.now
-      str_host = "http://#{host}:#{port}".colorize(:light_cyan).underline
+      ssl_enabled = ssl_key_file && ssl_cert_file
```
Comment (eliasjpr) [inline src/amber.cr]: 👍

File: src/amber.cr [hunk 2]
```diff
@@ -98,6 +102,13 @@ module Amber
 
       server = HTTP::Server.new(host, port, handler)
 
+      if ssl_enabled
```
Comment (eliasjpr) [inline src/amber.cr]: maybe we an refactor all the SSL stuff into a method?

Comment (elorest) [inline src/amber.cr]: Agreed

**Comment by c910335** [comment]
`key.pem` and `cert.pem` were generated by `openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -nodes` with all fields empty.

File: spec/amber/ssl/cert.pem [hunk 3]
```diff
@@ -0,0 +1,31 @@
+-----BEGIN CERTIFICATE-----
```
Comment (fridgerator) [inline spec/amber/ssl/cert.pem]: @elorest these are just for specs

Comment (c910335) [inline spec/amber/ssl/cert.pem]: @fridgerator absolutely

**Pull request merged**